src/transaction/NewTransaction.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import React, { useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import '../assets/styles/Table.scss';
+import sideimag from '../assets/images/gg.png';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import * as TransactionsAPI from '../dashboard/transactions-api';
+import { toast } from 'react-toastify';
+import { Calendar } from 'primereact/calendar';
+import 'react-toastify/dist/ReactToastify.css';
+import { v4 as uuidv4 } from 'uuid';
+
+const NewTransaction = () => {
+  const validationSchema = Yup.object().shape({
+    name: Yup.string().required('Name is required').min(3, 'Name must be at least 3 characters'),
+    category: Yup.string().required('Category is required'),
+    amount: Yup.number().required('Amount is required').typeError('Amount must be a number').positive('Amount must be positive'),
+    date: Yup.date().required('Date is required'),
+  });
+
+  const initialValues = {
+    name: '',
+    date: new Date(),
+    category: '',
+    amount: 0,
+  };
+
+  const [successMessage] = useState('');
+  const navigate = useNavigate();
+
+  const onSubmit = async (values: { date: { toLocaleDateString: () => any; }; name: any; category: any; amount: any; }, { resetForm, setSubmitting }: any) => {
+    const formattedDate = values.date.toLocaleDateString();
+    const newTransaction = {
+      name: values.name,
+      category: values.category,
+      date: formattedDate,
+      amount: values.amount,
+      id: parseInt(uuidv4(), 16)
```

Comment:
you can generate a random number with pure js script:
`Math. floor(Math. random()*10000) + 1`
or you can get the last index and increment it ;)